### ISSUE TITLE
Test agent will not survive restart of MySQL

### ISSUE THREAD

**Issue by matsduf** [zonemaster/zonemaster-backend#878]
Backend is running with MySQL backend and code from commit 60da83a. Everything works well and no tests are running. Then MySQL is restarted:
```
2021-09-29T09:45:15Z [3994] WARNING - DBD::mysql::db selectrow_array failed: MySQL server has gone away at /usr/local/lib/perl5/site_perl/Zonemaster/Backend/DB.pm line 238.
2021-09-29T09:45:15Z [3994] CRITICAL - DBD::mysql::db selectrow_array failed: MySQL server has gone away at /usr/local/lib/perl5/site_perl/Zonemaster/Backend/DB.pm line 238.
2021-09-29T09:45:15Z [3994] NOTICE - Daemon terminating
```

After that, the Test Agent is not running.

**Comment by pnax** [comment]
Doing some tests, it appears this also happens when restarting PostgreSQL.

**Comment by pnax** [comment]
My understanding is that the TestAgent ping the database. If there is no response to the ping, then the agent tries to open a connection to the database. However, the current code is not protected against connection failure because of a closed database.

It would indeed be nice to log the failure to connect without killing the agent.

Some pointers to some code involved:
https://github.com/zonemaster/zonemaster-backend/blob/b0e46a8477c7da5c150b9babf6acb110fab8143c/lib/Zonemaster/Backend/DB.pm#L76-L81
https://github.com/zonemaster/zonemaster-backend/blob/b0e46a8477c7da5c150b9babf6acb110fab8143c/lib/Zonemaster/Backend/DB.pm#L356-L364

**Comment by pnax** [comment]
Resolved with #955
